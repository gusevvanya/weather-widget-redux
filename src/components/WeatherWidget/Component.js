import React from 'react';

import SearchForm from '../SearchForm';
import CitiesList from '../CitiesList/CitiesList';
import WeatherContent from '../WeatherContent';
import MessageList from '../MessageList/MessageList';

const WeatherWidget = ({
  cities,
  currentCity,
  currentWeather,
  onSearchSubmit,
  onCityClick,
  onDeleteCityClick,
  messages,
  onClearMessageClick,
  children
}) => {
  return (
    <div className="weatherWidget">

      <SearchForm 
        onSearchSubmit={onSearchSubmit}
      />

      {cities && 
        <CitiesList 
          cities={cities} 
          currentCity={currentCity}
          onCityClick={onCityClick}
          onDeleteCityClick={onDeleteCityClick}
        />
      }

      {currentCity  && 
        <WeatherContent 
          weather={currentWeather}
          currentCity={currentCity}
        />
      }

      {messages &&
        <MessageList
          messages={messages}
          onClearMessageClick={onClearMessageClick}
        />
      }

      {children}

    </div>
  );
};

export default WeatherWidget;
