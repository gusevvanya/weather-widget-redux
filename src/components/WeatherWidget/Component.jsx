import React from 'react';
import PropTypes from 'prop-types';

import SearchForm from '../SearchForm';
import CitiesList from '../CitiesList/CitiesList';
import WeatherContent from '../WeatherContent';
import MessageList from '../MessageList/MessageList';

const WeatherWidget = ({
  cities,
  currentWeather,
  onSearchSubmit,
  onCityClick,
  onDeleteCityClick,
  messages,
  onClearMessageClick,
  children,
  currentCity1,
}) => {
  const currentCity = currentCity1;

  return (
    <div className="weatherWidget">
      <SearchForm
        onSearchSubmit={onSearchSubmit}
      />
      <CitiesList
        cities={cities}
        currentCity={currentCity}
        onCityClick={onCityClick}
        onDeleteCityClick={onDeleteCityClick}
      />
      <WeatherContent
        weather={currentWeather}
        currentCity={currentCity}
      />

      <MessageList
        messages={messages}
        onClearMessageClick={onClearMessageClick}
      />
      {children}
    </div>
  );
};

WeatherWidget.propTypes = {
  cities: PropTypes.arrayOf(PropTypes.string).isRequired,
  currentWeather: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]).isRequired,
  onSearchSubmit: PropTypes.func.isRequired,
  onCityClick: PropTypes.func.isRequired,
  onDeleteCityClick: PropTypes.func.isRequired,
  onClearMessageClick: PropTypes.func.isRequired,
  currentCity1: PropTypes.string.isRequired,
  messages: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default WeatherWidget;
