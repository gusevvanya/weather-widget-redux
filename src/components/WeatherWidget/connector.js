import { connect } from 'react-redux';
import { deleteCity, fetchCity } from '../../actions/cities';
import { deleteMessages } from '../../actions/ui/messages';

const mapStateToProps = state => ({
  cities: state.cities,
  currentWeather: state.currentWeather,
  messages: state.ui.messages,
  currentCity1: state.ui.currentCity,
});

const mapDispatchToProps = dispatch => ({
  onDeleteCityClick: (city) => {
    dispatch(deleteCity(city));
  },
  onCityClick: (city) => {
    dispatch(fetchCity(city));
  },
  onSearchSubmit: (city) => {
    dispatch(fetchCity(city));
  },
  onClearMessageClick: () => {
    dispatch(deleteMessages());
  },
});

export default WrappedComponent => connect(mapStateToProps, mapDispatchToProps)(WrappedComponent);
