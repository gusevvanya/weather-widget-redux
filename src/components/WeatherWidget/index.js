import {compose} from 'redux';

import addIntervalUpdateWeather from '../../HOC/addIntervalUpdateWeather';
import connector from './connector';
import Component from './Component';

const WrappedComponent = compose(
	connector,
	addIntervalUpdateWeather
)(Component);

export default WrappedComponent;