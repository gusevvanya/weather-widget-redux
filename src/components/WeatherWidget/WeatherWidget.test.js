import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import WeatherWidget from './Component';

Enzyme.configure({ adapter: new Adapter() });

describe('WeatherWidget', () => {
  let props;
  let weatherWidgetShallow;

  beforeAll(() => {
    props = {
      cities: ['London', 'Kharkov'],
      currentWeather: '21.25',
      onSearchSubmit: jest.fn(),
      onCityClick: jest.fn(),
      onDeleteCityClick: jest.fn(),
      onClearMessageClick: jest.fn(),
      currentCity1: 'London',
      messages: [{ value: 'New massage', type: 'success' }],
    };
  });

  beforeEach(() => {
    jest.clearAllMocks();
    weatherWidgetShallow = Enzyme.mount(<WeatherWidget {...props} />)
  });

  it('renders correctly', () => {
    expect(weatherWidgetShallow.debug()).toMatchSnapshot();
  });
});
