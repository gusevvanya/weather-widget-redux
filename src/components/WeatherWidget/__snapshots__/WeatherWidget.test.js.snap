// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherWidget renders correctly 1`] = `
"<WeatherWidget cities={{...}} currentWeather=\\"21.25\\" onSearchSubmit={[Function]} onCityClick={[Function]} onDeleteCityClick={[Function]} onClearMessageClick={[Function]} currentCity1=\\"London\\" messages={{...}}>
  <div className=\\"weatherWidget\\">
    <SearchForm onSearchSubmit={[Function]}>
      <form className=\\"search\\" onSubmit={[Function]}>
        <TextField onTextFieldChange={[Function]} value=\\"\\">
          <input type=\\"text\\" value=\\"\\" onChange={[Function]} />
        </TextField>
        <Button>
          <button>
            Search
          </button>
        </Button>
      </form>
    </SearchForm>
    <CitiesList cities={{...}} currentCity=\\"London\\" onCityClick={[Function]} onDeleteCityClick={[Function]}>
      <div className=\\"cities\\">
        <City name=\\"London\\" className=\\"city active\\" onCityClick={[Function]} onDeleteCityClick={[Function]}>
          <a className=\\"city active\\" onClick={[Function]}>
            <button className=\\"close\\" onClick={[Function]}>
              x
            </button>
            London
          </a>
        </City>
        <City name=\\"Kharkov\\" className=\\"city\\" onCityClick={[Function]} onDeleteCityClick={[Function]}>
          <a className=\\"city\\" onClick={[Function]}>
            <button className=\\"close\\" onClick={[Function]}>
              x
            </button>
            Kharkov
          </a>
        </City>
      </div>
    </CitiesList>
    <WeatherContent weather=\\"21.25\\" currentCity=\\"London\\">
      <div className=\\"weather\\">
        London
        <br />
        21.25
         Â°F
      </div>
    </WeatherContent>
    <MessageList messages={{...}} onClearMessageClick={[Function]}>
      <div className=\\"messageList\\">
        <ul>
          <MessageItem value=\\"New massage\\" type=\\"success\\">
            <li className=\\"success\\">
              New massage
            </li>
          </MessageItem>
        </ul>
        <button onClick={[Function]}>
          x
        </button>
      </div>
    </MessageList>
  </div>
</WeatherWidget>"
`;
